### ISSUE TITLE
Crash when setting :insert_mode to :button

### ISSUE THREAD

**Issue by jeffcarbs** [bmichotte/ProMotion-XLForm#23]
I'm trying to set the insert mode to `button` instead of `last_row` via https://github.com/bmichotte/ProMotion-XLForm/blob/master/lib/ProMotion/XLForm/xl_form_patch.rb#L117

However, when I set `insert_mode` to `:button` I get the following crash:

```
ProMotion-XLForm[2468:8710476] *** Terminating app due to uncaught exception 'TypeError', reason: 'xl_form_patch.rb:82:in `section:': can't convert nil into Integer (TypeError)
    from xl_form_section_builder.rb:5:in `create_section:'
    from xl_form.rb:21:in `block in build'
    from xl_form.rb:20:in `build'
    from xl_form_screen.rb:64:in `update_form_data'
    from xl_form_screen.rb:10:in `viewDidLoad'
'
*** First throw call stack:
(
    0   CoreFoundation                      0x0000000102f7ef45 __exceptionPreprocess + 165
    1   libobjc.A.dylib                     0x000000010050adeb objc_exception_throw + 48
    2   ProMotion-XLForm                    0x000000010025e320 _ZL10__vm_raisev + 400
    3   ProMotion-XLForm                    0x000000010025e450 rb_vm_raise + 208
    4   ProMotion-XLForm                    0x0000000100159f89 rb_exc_raise + 9
    5   ProMotion-XLForm                    0x0000000100153e15 rb_raise + 245
    6   ProMotion-XLForm                    0x000000010018ef47 rb_convert_to_integer + 775
    7   ProMotion-XLForm                    0x000000010003cb0b vm_rval_to_ulong_long + 475
    8   ProMotion-XLForm                    0x00000001000be3e9 __unnamed_100 + 25
    9   ProMotion-XLForm                    0x000000010001f6e6 -[XLFormSectionDescriptor canInsertUsingButton] + 44
    10  ProMotion-XLForm                    0x000000010001de53 -[XLFormSectionDescriptor initWithTitle:sectionOptions:sectionInsertMode:] + 131
    11  ProMotion-XLForm                    0x000000010001e166 +[XLFormSectionDescriptor formSectionWithTitle:sectionOptions:sectionInsertMode:] + 90
    12  ProMotion-XLForm                    0x00000001000bff92 __unnamed_253 + 162
    13  ProMotion-XLForm                    0x000000010022f424 rb_vm_dispatch + 6196
    14  ProMotion-XLForm                    0x000000010003326c vm_dispatch + 1436
    15  ProMotion-XLForm                    0x00000001000be037 rb_scope__section:__ + 647
    16  ProMotion-XLForm                    0x000000010022f8bc rb_vm_dispatch + 7372
    17  ProMotion-XLForm                    0x000000010003326c vm_dispatch + 1436
    18  ProMotion-XLForm                    0x00000001000c8c6a rb_scope__create_section:__ + 202
    19  ProMotion-XLForm                    0x000000010022f8bc rb_vm_dispatch + 7372
    20  ProMotion-XLForm                    0x000000010003326c vm_dispatch + 1436
    21  ProMotion-XLForm                    0x00000001000cb039 rb_scope__build__block__ + 121
    22  ProMotion-XLForm                    0x0000000100230c8f _ZL13vm_block_evalP7RoxorVMP11rb_vm_blockP13objc_selectormiPKm + 1439
    23  ProMotion-XLForm                    0x0000000100230e90 rb_vm_yield_args + 64
    24  ProMotion-XLForm                    0x00000001002248cb rb_yield + 59
    25  ProMotion-XLForm                    0x000000010011c73d rary_each + 77
    26  ProMotion-XLForm                    0x000000010022f8bc rb_vm_dispatch + 7372
    27  ProMotion-XLForm                    0x000000010003326c vm_dispatch + 1436
    28  ProMotion-XLForm                    0x00000001000caf7d rb_scope__build__ + 637
    29  ProMotion-XLForm                    0x000000010022f8bc rb_vm_dispatch + 7372
    30  ProMotion-XLForm                    0x000000010003326c vm_dispatch + 1436
    31  ProMotion-XLForm                    0x00000001000d3afc rb_scope__update_form_data__ + 812
    32  ProMotion-XLForm                    0x000000010022f8bc rb_vm_dispatch + 7372
    33  ProMotion-XLForm                    0x000000010003326c vm_dispatch + 1436
    34  ProMotion-XLForm                    0x00000001000d2d4e rb_scope__viewDidLoad__ + 238
    35  ProMotion-XLForm                    0x00000001000d35ad __unnamed_18 + 13
    36  UIKit                               0x0000000100dc4cc4 -[UIViewController loadViewIfRequired] + 1198
    37  UIKit                               0x0000000100e090ee -[UINavigationController _layoutViewController:] + 54
    38  UIKit                               0x0000000100e099c2 -[UINavigationController _updateScrollViewFromViewController:toViewController:] + 462
    39  UIKit                               0x0000000100e09b34 -[UINavigationController _startTransition:fromViewController:toViewController:] + 126
    40  UIKit                               0x0000000100e0ad8d -[UINavigationController _startDeferredTransitionIfNeeded:] + 890
    41  UIKit                               0x0000000100e0bcea -[UINavigationController __viewWillLayoutSubviews] + 57
    42  UIKit                               0x0000000100fb1c85 -[UILayoutContainerView layoutSubviews] + 248
    43  UIKit                               0x00000001139c9c30 -[UILayoutContainerViewAccessibility layoutSubviews] + 43
    44  UIKit                               0x0000000100ce6e40 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 710
    45  QuartzCore                          0x00000001028e859a -[CALayer layoutSublayers] + 146
    46  QuartzCore                          0x00000001028dce70 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 366
    47  QuartzCore                          0x00000001028dccee _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 24
    48  QuartzCore                          0x00000001028d1475 _ZN2CA7Context18commit_transactionEPNS_11TransactionE + 277
    49  QuartzCore                          0x00000001028fec0a _ZN2CA11Transaction6commitEv + 486
    50  QuartzCore                          0x00000001028ff37c _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 92
    51  CoreFoundation                      0x0000000102eaa947 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
    52  CoreFoundation                      0x0000000102eaa8b7 __CFRunLoopDoObservers + 391
    53  CoreFoundation                      0x0000000102e9fe2c CFRunLoopRunSpecific + 524
    54  UIKit                               0x0000000100c2d4f5 -[UIApplication _run] + 402
    55  UIKit                               0x0000000100c3230d UIApplicationMain + 171
    56  ProMotion-XLForm                    0x000000010003de5f main + 111
    57  libdyld.dylib                       0x000000010427192d start + 1
)
libc++abi.dylib: terminating with uncaught exception of type TypeError
```

https://github.com/jcarbo/ProMotion-XLForm/commit/5de974bc2a3bb0c05b9cf58241321cf66678939e (just adding the `:insert_mode` option) causes a crash.

Some other notes:
- Setting `insert_mode` to `:last_row` works as expected.
- Setting `insert_mode` to `XLFormSectionInsertModeButton` causes the same crash.

**Comment by jeffcarbs** [comment]
Update: I fixed the bug. I'll have a PR up shortly with an explanation.

**Comment by bmichotte** [comment]
Nice, I'm waiting the explanation, I don't understand why it crash

**Comment by bmichotte** [comment]
Closed by #24
